src/Middleware/Spa/SpaServices.Extensions/src/Proxying/SpaProxy.cs

```diff
@@ -70,7 +70,7 @@ public static HttpClient CreateHttpClientForProxy(TimeSpan requestTimeout)
         var baseUri = await baseUriTask;
         var targetUri = new Uri(
             baseUri,
-            context.Request.Path + context.Request.QueryString);
+            "/" + (baseUri.AbsolutePath + context.Request.Path).TrimStart('/') + context.Request.QueryString);
```

Comment:
Sorry, Trim wasn't the solution recommended in the issue, that would remove multipe slashes and not preserve the user intent.

> A period . needs to be prefixed to the relative URI to avoid this behaviour.

I didn't realize there was a competing PR here https://github.com/dotnet/aspnetcore/pull/43663/files. They were going to conflict regardless. You can revert to your original fix and we'll sort it out on the other PR.

Reply:
@Tratcher that PR would also fix #31229 if the base URL has a slash as a suffix like:
`spa.UseProxyToSpaDevelopmentServer("http://localhost:3001/spa/");`

If we want to support the case when the URL is given like without the slash at the end (http://localhost:3001/spa), then we should go with this:
```csharp
        var targetUri = new Uri(
            baseUri.ToString().EndsWith("/") ? baseUri : new Uri(baseUri + "/"),
            "." + context.Request.Path + context.Request.QueryString);
```